client/pages/clubs/[id].tsx

```diff
@@ -1,22 +1,251 @@
+  margin: 0 auto 100px;
+  width: 65%;
+
+  @media ${device.mobileL} {
+    width: 85%;
+  }
+`
 
 export default function ClubDetail() {
-  const { clubs } = useAppContext();
-  const router = useRouter();
+  const { clubs } = useAppContext()
+  const router = useRouter()
 
-  const { id } = router.query;
-  if (typeof id !== 'string') return null;
+  const { id } = router.query
+  if (typeof id !== 'string') return null
 
-  const { name, description } = clubs.get(parseInt(id));
+  const club = clubs.get(parseInt(id))
 
   return (
     <div>
-      <h1>Club Details</h1>
-      <p>{id}</p>
-      <p>{name}</p>
-      <p>{description}</p>
+      <Banner backgroundImageUrl={club.backgroundImageURL} />
+      <Container>
+        <ClubInfo club={club} />
+        <EventsHostedByClub events={club.events} />
+      </Container>
     </div>
-  );
+  )
+}
+
+const BackArrow = styled.img`
+  margin-top: 64px;
+`
+
+const Logo = styled.img`
+  border-radius: 100%;
+  display: block;
+  margin-top: 40px;
+  height: 100px;
+  width: 100px;
+`
+
+const ClubName = styled(PageTitle)`
+  margin-bottom: 0;
+  margin-top: 32px;
+`
+
+const Links = styled.div`
+  display: flex;
+  margin-top: 16px;
+`
+
+const SocialLink = styled.div`
+  text-decoration: underline;
+`
+
+const ClubMetaData = styled.div`
+  align-items: center;
+  display: flex;
+  justify-content: space-between;
+  margin-top: 8px;
+`
+
+const MemberCount = styled.div`
+  font-style: italic;
+  font-weight: ${fontWeight.bold};
+`
+
+const CategoriesWrapper = styled.div`
+  display: flex;
+`
+
+const Category = styled.p`
+  border: 1px solid black;
+  border-radius: 15px;
+  font-size: 12px;
+  margin-left: 5px;
+
+  padding: 5px 10px;
+  @media ${device.mobileS} {
+    padding: 2px 5px;
+  }
+`
+
+const ClubDescription = styled.p`
+  margin-top: 24px;
+`
+
+interface ClubInfoProps {
+  club: Club
 }
+
+const ClubInfo = ({ club }: ClubInfoProps) => {
+  const { name, description, iconURL, tags } = club
+  const links = [club.facebookLink, club.twitterLink, club.instagramLink, club.websiteLink]
+  const linksThatExist = links.filter((l) => l !== undefined)
+
+  return (
+    <div>
+      <BackArrow src="/back-arrow.svg" width="28px" />
+
+      <Logo src={iconURL} alt="" />
+      <ClubName> {name}</ClubName>
+
+      <Links>
+        {linksThatExist.map((l) => (
+          <SocialLink>{l}</SocialLink>
+        ))}
+      </Links>
+
+      <ClubMetaData>
+        <MemberCount>20-30 members</MemberCount>
+        <Categories tags={tags} />
+        {/* TODO: color code the tags */}
+      </ClubMetaData>
+
+      <ClubDescription>{description}</ClubDescription>
+    </div>
+  )
+}
+
+const Categories = ({ tags }: { tags: Array<string> }) => (
+  <CategoriesWrapper>
+    {tags.map((t) => (
+      <Category>{t}</Category>
+    ))}
+  </CategoriesWrapper>
+)
+
+const EventsTitle = styled.h2`
+  margin-top: 32px;
+`
+
+const EventCardDetails = styled.div`
+  margin: 0 20px 20px;
```

Comment:
personally I prefer explicit margin-left/right/top/bottom instead of the shorthand, because I still haven't memorized the order

Reply:
Changed